modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItSqlMultiStatementTxTest.java

```diff
@@ -230,22 +230,18 @@ void dmlScriptRollsBackImplicitly() throws InterruptedException {
 
     @Test
     void openedScriptTransactionRollsBackOnError() {
-        {
-            AsyncSqlCursor<InternalSqlRow> cursor = runScript(
-                    "START TRANSACTION READ WRITE;"
-                    + "INSERT INTO test VALUES(2);"
-                    + "INSERT INTO test VALUES(2/0);"
-                    + "SELECT 1;"
-                    + "COMMIT;"
-            );
+        String script = "START TRANSACTION READ WRITE;"
+                + "INSERT INTO test VALUES(2);"
+                + "INSERT INTO test VALUES(2/0);"
```

Comment:
what is the expected behavior if SELECT will raise an error?
Should script be executed completely?

From my experiments (may be wrong) I see that sometime the script completes with error, but sometimes without, my be we need to not interrupt script execution in case of errors from  SELECT statements at all.

Reply:
> what is the expected behavior if SELECT will raise an error?

as far as I see, it depends... If exception is raised during cursor preparation, then an entire script will be interrupted. But now, data fetch is not part of the cursor preparation for SELECT statement, thus, script won't be interrupted, if it works in autoCommit=true mode